File: src/Java/Comment.java
package Java;

import java.util.Date;

public class Comment {


    private int id;
    private Date date;
    private User author;
    private String text;
    private Post post;
    private String editorialStatus;


    public Comment(){
    }

    public Comment(int id, String text, User author, Post post){
        this.id = id;
        this.text = text;
        this.author = author;
        this.date = new Date();
    }

    // Metodos Getters
    public int getId(){
        return this.id;
    }

    public Date getDate(){
        return this.date;
    }

    public User getAuthor(){
        return this.author;
    }

    public String getText(){
        return this.text;
    }
    
    public String getEditorialStatus(){
        return this.editorialStatus;
    }
    
    public Post getPost(){
        return this.post;
    }
    
    // Metodos Setters
    public void setText(String text){
        this.text = text;
    }

    public void setEditorialStatus(String editorialStatus){
        this.editorialStatus = editorialStatus;
    }   

    public void setAuthor(User author){
        this.author = author;
    }

    public void setDate(Date date){
        this.date = date;
    }

    public void setPost(Post post){
        this.post = post;
    }

    @Override 
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("-----------------------------------\n");
        sb.append("Id: " + this.id + "\n");
        sb.append("Autor: " + this.author.getUsername() + "\n");
        sb.append("Fecha: " + this.date + "\n");
        sb.append("Texto: " + this.text + "\n");
        sb.append("-------------------------------------");
        return sb.toString();
    }

}

File: src/C#/networking-app/MyApplication/Comment.cs
using System;
using static System.DateTime;
namespace MyApplication.Models
{
	public class Comment 
	{
		private int id;
		private static int coutsId;
		private DateTime date;
		private User author;
		private string content;
		private Post post;
		private string editorialStatus;

		
		public Comment(int id, string content, User author, Post post)
		{
			this.id = ++Comment.coutsId;
			this.content = content;
			this.author = author;
			this.date = DateTime.Now;
			this.editorialStatus = "";
			this.post = post;
		}
		
		// Metodos Getters y Setters
		public int GetId() => this.id;
		
		public string GetContent() => this.content;
		public User GetAuthor() => this.author;
		public DateTime GetDate() => this.date;
		public string GetEditorialStatus() => this.editorialStatus;
		public void SetContent(string content) => this.content = content;
		public void SetAuthor(User author) => this.author = author;
		public void SetDate(DateTime date) => this.date = date;
		public void SetEditorialStatus(string editorialStatus) => this.editorialStatus = editorialStatus;
		public override string ToString(){
			return $"Id: {this.id}\n" +
				   $"Autor: {this.author.Name}\n" +
				   $"Fecha: {this.date}\n" +
				   $"Texto: {this.content}\n" +
				   $"Estado de editorial: {this.editorialStatus}\n";
		}
	}
}
